src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,286 @@
+from __future__ import (division, print_function)
+
+import json
+import logging
+
+from WMCore.Services.Service import Service
+from pprint import pformat
+
+
+class DDMReqTemplate(dict):
+    """
+    A simple class defining a set of request templates for DDM, with default
+    structures representing the different POST calls expected by the modules
+    here:
+        https://github.com/SmartDataProjects/dynamo/tree/master/lib/web/modules
+    """
+
+    def __init__(self, api=None, **kwargs):
+        """
+        api:
+            String representing the type of request to be sent. In case the api
+            is missing or unrecognized the created object will be an ordinary
+            dictionary.
+        example:
+            ddmReq = DDMReqTemplate('copy',
```

Comment:
Should we validate the data type for those keyword arguments as well? For instance, can I set `site` to a string, or it has to be a list?

Reply:
In case you are not going for an aggregated Request it can be a string too. It will be sent to DDM as it is. And since DDM can digest both - lists and strings, it really doesn't matter. But I am more for sticking to a common practice everywhere, and would say we better send it always as a list. So a check here doesn't hurt.

Reply:
Done. Every parameter in the template is now checked against its expected type and ValueError is raised in case of mismatch.